modules/plugins/youtube.go

```diff
@@ -199,93 +218,230 @@ func (yt *YouTube) actionChannel(args []string, in *discordgo.Message, out **dis
+		content.ID = yc.Snippet.ChannelId
+	}
+
+	if content.ID == "" || content.Name == "" {
+		*out = yt.newMsg("plugins.youtube.channel-not-found")
+		return yt.actionFinish
+	}
+
+	// fill db entry
+	entry := DB_Youtube_Entry{
+		ServerID:                dc.GuildID,
+		ChannelID:               dc.ID,
+		NextCheckTime:           time.Now().Unix(),
+		LastSuccessfulCheckTime: time.Now().Unix(),
+
+		ContentType: "channel",
+		Content:     content,
+	}
+
+	// insert entry into the db
+	_, err = yt.createEntry(entry)
+	if err != nil {
+		yt.logger().Error(err)
+		*out = yt.newMsg(err.Error())
+		return yt.actionFinish
+	}
+
+	yt.quota.IncEntryCount()
+
+	*out = yt.newMsg("Added youtube channel `" + yc.Snippet.ChannelTitle + "` to the discord channel <#" + dc.ID + ">")
 	return yt.actionFinish
 }
 
+// _yt channel delete <channel id> <discord channel>
 func (yt *YouTube) actionDeleteChannel(args []string, in *discordgo.Message, out **discordgo.MessageSend) youtubeAction {
-	if len(args) < 4 {
+	if len(args) < 3 {
 		*out = yt.newMsg("bot.arguments.too-few")
 		return yt.actionFinish
 	}
 
-	// _yt video delete <video id/link> <discord channel>
+	if helpers.IsMod(in) == false {
+		*out = yt.newMsg("mod.no_permission")
+		return yt.actionFinish
+	}
 
-	*out = yt.newMsg("bot.arguments.invalid")
+	n, err := yt.deleteEntry(args[2])
+	if err != nil {
+		yt.logger().Error(err)
+		*out = yt.newMsg(err.Error())
+		return yt.actionFinish
+	}
+
+	if n < 1 {
+		*out = yt.newMsg("plugins.youtube.channel-delete-not-found-error")
+		return yt.actionFinish
+	}
+
+	yt.quota.DecEntryCount()
+
+	*out = yt.newMsg("Delete channel, ID: " + args[2])
 	return yt.actionFinish
 }
 
+// _yt channel list
 func (yt *YouTube) actionListChannel(args []string, in *discordgo.Message, out **discordgo.MessageSend) youtubeAction {
+	if helpers.IsMod(in) == false {
+		*out = yt.newMsg("mod.no_permission")
+		return yt.actionFinish
+	}
 
-	// _yt video list
+	ch, err := helpers.GetChannel(in.ChannelID)
+	if err != nil {
+		*out = yt.newMsg(err.Error())
+		return yt.actionFinish
+	}
 
-	*out = yt.newMsg("bot.arguments.invalid")
+	entries, err := yt.readEntries(map[string]interface{}{
+		"content_type": "channel",
+		"server_id":    ch.GuildID,
+	})
+	if err != nil {
+		yt.logger().Error(err)
+		*out = yt.newMsg(err.Error())
+		return yt.actionFinish
+	}
+
+	if len(entries) < 1 {
+		*out = yt.newMsg("No entries")
+		return yt.actionFinish
+	}
+
+	msg := ""
+	for _, e := range entries {
+		c, err := e.getChannelContent()
+		if err != nil {
+			yt.logger().WithFields(logrus.Fields{
+				"id": e.ID,
+			}).Error(err)
+			continue
+		}
+		msg += fmt.Sprintf("`%s`: Youtube channel name `@%s` posting to <#%s>\n", e.ID, c.Name, e.ChannelID)
+	}
+
+	for _, resultPage := range helpers.Pagify(msg, "\n") {
+		_, err := cache.GetSession().ChannelMessageSend(in.ChannelID, resultPage)
+		helpers.Relax(err)
+	}
+
+	msg = fmt.Sprintf("Found **%d** Youtube channel in total.", len(entries))
```

Comment:
Please edit `channel` to `channel(s)` to make the text work with multiple channels.

Reply:
changed, and added to i18n json.